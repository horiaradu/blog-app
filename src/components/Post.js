import React, { Component } from 'react';
import '../css/posts.css';
import Comments from './Comments';
import BlogForm from './BlogForm';
import EditEntry from './EditEntry';
import DeleteEntry from './DeleteEntry';

class Post extends Component {
  state = {
    isEditModeOn: false,
    isUser: false
  };
  changeState = () => {
    this.setState({ isEditModeOn: !this.state.isEditModeOn });
  };
  isAuthor = () => {
    if (this.props.currentUser.userId === this.props.post.entry.userId) {
      return true;
    }
    return false;
  };
  getUserName = () => {
    const userName = this.props.users.find(user => {
      return user.userId === this.props.post.entry.userId;
    });
    return userName;
  };

  render() {
    const { title, body, tags, userId } = this.props.post.entry;

    const entryUuid = this.props.post.entry.uuid;
    const currentUser = this.props.currentUser;
    const userName = this.getUserName();
    return (
      <div className="blogSeparator">
        {this.state.isEditModeOn ? (
          <BlogForm
            onUpdateClick={this.changeState}
            entry={this.props.post.entry}
            entryUuid={entryUuid}
            userId={userId}
          />
        ) : (
          <div>
            <div className="postTitle">
              <span>
                <h2>{title}</h2>
              </span>
              <span>
                {this.isAuthor() && (
                  <span>
                    <EditEntry onEditClick={this.changeState} entryUuid={entryUuid} />
                    <DeleteEntry onEditClick={this.changeState} entryUuid={entryUuid} />
                  </span>
                )}
              </span>
            </div>
            <div className="postContent">
              {this.getUserName() && (
                <h4 key={entryUuid}>Created by - {`${userName.firstName} ${userName.lastName}`}</h4>
              )}

              <p>{body}</p>
              <ul className="ulStyle">
                {tags
                  ? tags.map(tag => {
                      return (
                        <li key={tag} className="tagStyle">
                          {tag}
                        </li>
                      );
                    })
                  : null}
              </ul>
              <Comments
                comments={this.props.post.comments}
                entryUuid={this.props.post.entry.uuid}
                currentUser={currentUser}
              />
            </div>
          </div>
        )}
      </div>
    );
  }
}

export default Post;
