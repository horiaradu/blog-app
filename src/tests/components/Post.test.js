import React from 'react';
import { shallow } from 'enzyme';
import Post from '../../components/Post';
import toJSON from 'enzyme-to-json';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
describe('Post Component', () => {
  it('should render Post component correctly', () => {
    const blog = {
      comments: [],
      entry: {
        title: 'title',
        body: 'body text',
        tags: ['tag1', 'tag2'],
        uuid: '123456'
      }
    };
    const currentUser = {
      firstName: 'Tony',
      userId: '999'
    };
    const users = [
      {
        firstName: 'Tony',
        userId: '999'
      }
    ];

    const wrapper = shallow(<Post key={blog.entry.uuid} post={blog} currentUser={currentUser} users={users} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(blog.entry.tags[0]).toBe('tag1');
  });
  it('should not render tags if no tags provided', () => {
    const blog = {
      comments: [],
      entry: {
        title: 'title',
        body: 'body text',
        tags: [],
        uuid: '123456'
      }
    };
    const currentUser = {
      firstName: 'Tony',
      userId: '999'
    };
    const users = [
      {
        firstName: 'Tony',
        userId: '999'
      }
    ];

    const wrapper = shallow(<Post key={blog.entry.uuid} post={blog} currentUser={currentUser} users={users} />);
    expect(wrapper.find('li').length).toBe(0);
  });
});
