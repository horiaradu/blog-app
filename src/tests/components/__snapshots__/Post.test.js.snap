// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post Component should render Post component correctly 1`] = `
<div
  className="blogSeparator"
>
  <div>
    <div
      className="postTitle"
    >
      <span>
        <h2>
          title
        </h2>
      </span>
      <span />
    </div>
    <div
      className="postContent"
    >
      <p>
        body text
      </p>
      <ul
        className="ulStyle"
      >
        <li
          className="tagStyle"
          key="tag1"
        >
          tag1
        </li>
        <li
          className="tagStyle"
          key="tag2"
        >
          tag2
        </li>
      </ul>
      <Connect(Comments)
        comments={Array []}
        currentUser={
          Object {
            "firstName": "Tony",
            "userId": "999",
          }
        }
        entryUuid="123456"
      />
    </div>
  </div>
</div>
`;
